test/object-store/migrations.cpp

```diff
@@ -1938,248 +1952,302 @@ TEST_CASE("migration: Additive") {
-    auto realm = Realm::get_shared_realm(config);
-    realm->update_schema(schema);
 
-    SECTION("can add new properties to existing tables") {
-        REQUIRE_NOTHROW(realm->update_schema(add_property(schema, "object", {"value 3", PropertyType::Int})));
-        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object")->get_column_count() == 3);
-    }
+    std::vector<SchemaMode> additive_modes = {SchemaMode::AdditiveDiscovered, SchemaMode::AdditiveExplicit};
 
-    SECTION("can add new tables") {
-        REQUIRE_NOTHROW(realm->update_schema(add_table(schema, {"object 2",
-                                                                {
-                                                                    {"value", PropertyType::Int},
-                                                                }})));
-        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object"));
-        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object 2"));
-    }
+    for (auto mode : additive_modes) {
+        config.schema_mode = mode;
+        auto realm = Realm::get_shared_realm(config);
+        realm->update_schema(schema);
+        std::string mode_string = util::format(
+            " with mode: %1", mode == SchemaMode::AdditiveDiscovered ? "AdditiveDiscovered" : "AdditiveExplicit");
 
-    SECTION("cannot change existing table type") {
-        REQUIRE_THROWS(realm->update_schema(set_embedded(schema, "object", true)));
-    }
+        DYNAMIC_SECTION("can add new properties to existing tables" << mode_string)
+        {
+            REQUIRE_NOTHROW(realm->update_schema(add_property(schema, "object", {"value 3", PropertyType::Int})));
+            REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object")->get_column_count() == 3);
+        }
 
-    SECTION("indexes are updated when schema version is bumped") {
-        auto table = ObjectStore::table_for_object_type(realm->read_group(), "object");
-        auto col_keys = table->get_column_keys();
-        REQUIRE(table->has_search_index(col_keys[0]));
-        REQUIRE(!table->has_search_index(col_keys[1]));
+        DYNAMIC_SECTION("can add new tables" << mode_string)
+        {
+            REQUIRE_NOTHROW(realm->update_schema(add_table(schema, {"object 2",
+                                                                    {
+                                                                        {"value", PropertyType::Int},
+                                                                    }})));
+            REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object"));
+            REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object 2"));
+        }
 
-        REQUIRE_NOTHROW(realm->update_schema(set_indexed(schema, "object", "value", false), 1));
-        REQUIRE(!table->has_search_index(col_keys[0]));
+        DYNAMIC_SECTION("embedded orphan types" << mode_string)
```

Comment:
This is the only new test in this section, the other changes are just indentation and changing to dynamic sections.

Reply:
DYNAMIC_SECTION should be added to ForEachMacros in .clang-format so that it gets formatted as a control structure rather than just this weird thing sitting before a block.